src/main/java/vazkii/botania/common/item/ItemManaMirror.java

```diff
@@ -137,16 +135,10 @@ private static GlobalPos getBoundPos(ItemStack stack) {
 			return null;
 		}
 
-		Optional<GlobalPos> pos = GlobalPos.CODEC.parse(NBTDynamicOps.INSTANCE, ItemNBTHelper.get(stack, TAG_POS)).result();
-		if (!pos.isPresent()) {
-			return null;
-		}
-
-		BlockPos coords = pos.get().getPos();
-		if (coords.getY() == -1) {
-			return null;
-		}
-		return pos.get();
+		return GlobalPos.CODEC.parse(NBTDynamicOps.INSTANCE, ItemNBTHelper.get(stack, TAG_POS))
+				.result()
+				.filter(x -> x.getPos().getY() != -1)
```

Comment:
you inverted the condition

also name the lambda param `pos` since `x` is confusing

Reply:
filter returns a null optional when the condition is false. the condition is false when getY() == -1